Source/CarthageKit/Xcode.swift

```diff
@@ -664,6 +692,10 @@ private func build(sdk: SDK, with buildArgs: BuildArguments, in workingDirectory
 					return !dependencyCheckoutDir.hasSubdirectory(projectURL)
 				}
 				.collect()
+				.flatMap(.concat) { settings -> SignalProducer<[BuildSettings], CarthageError> in
+					let producers = settings.map { resolveSameTargetName(for: $0) }
+					return SignalProducer(value: producers).flatten().flatten(.concat).collect()
+				}
```

Comment:
I think moving this before the `.collect` would let you get rid of most of these operators.

```swift
.flatMap(.concat) { settings in resolveSameTargetName(for: settings) }
.collect()
```